test/test_DFT.py

```diff
@@ -159,25 +116,90 @@ def x_Ux_and_window(draw: Callable, ux_type: Optional[Union[np.ndarray, float]]
+    float_ux_strategy = random_not_negative_float_strategy()
     if ux_type == np.ndarray:
         uncertainty_strategy = full_ux_strategy
     elif ux_type == float:
         uncertainty_strategy = float_ux_strategy
     else:
         uncertainty_strategy = hst.one_of(float_ux_strategy, full_ux_strategy)
     Ux = draw(uncertainty_strategy)
-    window = draw(hnp.arrays(dtype=float, shape=dim))
+    window = draw(hypothesis_float_vector(length=dim))
 
     return {"x": x, "Ux": Ux, "window": window}
 
 
+@composite
+def random_vector_and_matching_random_square_matrix(
+    draw: Callable,
+) -> VectorAndCompatibleMatrix:
+    x = draw(hypothesis_float_vector())
+    A = draw(random_float_square_matrix_strategy(len(x)))
+    return VectorAndCompatibleMatrix(vector=x, matrix=A)
+
+
+@composite
+def random_vector_and_matrix_with_matching_number_of_rows(
+    draw: Callable,
+) -> VectorAndCompatibleMatrix:
+    x = draw(hypothesis_float_vector())
+    A = draw(random_float_matrix(number_of_rows=len(x)))
+    return VectorAndCompatibleMatrix(vector=x, matrix=A)
+
+
+@composite
+def random_vector_and_matrix_with_matching_number_of_columns(
+    draw: Callable,
+) -> VectorAndCompatibleMatrix:
+    x = draw(hypothesis_float_vector())
+    A = draw(random_float_matrix(number_of_cols=len(x)))
+    return VectorAndCompatibleMatrix(vector=x, matrix=A)
+
+
+class TestDFT:
+    def test_DFT_iDFT(self, multisine_testsignal):
+        """Test GUM_DFT and GUM_iDFT with noise variance as uncertainty"""
+        x, ux = multisine_testsignal
+        X, UX = GUM_DFT(x, ux ** 2)
+        xh, uxh = GUM_iDFT(X, UX)
+        assert_almost_equal(np.max(np.abs(x - xh)), 0)
+        assert_almost_equal(np.max(ux - np.sqrt(np.diag(uxh))), 0)
+
+    def test_DFT_iDFT_vector(self, multisine_testsignal):
+        """Test GUM_DFT and GUM_iDFT with uncertainty vector"""
+        x, ux = multisine_testsignal
+        ux = (0.1 * x) ** 2
+        X, UX = GUM_DFT(x, ux)
+        xh, uxh = GUM_iDFT(X, UX)
+        assert_almost_equal(np.max(np.abs(x - xh)), 0)
+        assert_almost_equal(np.max(np.sqrt(ux) - np.sqrt(np.diag(uxh))), 0)
+
+    def test_AmpPhasePropagation(self, multisine_testsignal):
+        """Test Time2AmpPhase and AmpPhase2Time with noise variance as uncertainty"""
+        testsignal, noise_std = multisine_testsignal
+        A, P, UAP = Time2AmpPhase(testsignal, noise_std ** 2)
+        x, ux = AmpPhase2Time(A, P, UAP)
+        assert_almost_equal(np.max(np.abs(testsignal - x)), 0)
+
+
+def test_compose_DFT_and_iDFT_with_full_covariance(multisine_testsignal, corrmatrix):
+    """Test GUM_DFT and GUM_iDFT with full covariance matrix"""
+    x, ux = multisine_testsignal
+    ux = np.ones_like(x) * 0.01 ** 2
+    cx = corrmatrix(0.95, len(x))
+    Ux = np.diag(ux)
+    Ux = Ux.dot(cx.dot(Ux))
+    X, UX = GUM_DFT(x, Ux)
+    xh, Uxh = GUM_iDFT(X, UX)
+    assert_almost_equal(np.max(np.abs(x - xh)), 0)
+    assert_almost_equal(np.max(Ux - Uxh), 0)
```

Comment:
What was the reasoning to not put `test_compose_DFT_and_iDFT_with_full_covariance` inside `class TestDFT`?

Reply:
Originally it was part of the class under the name of `test_DFT_iDFT_fullcov`. I touched this method, because in an intermediate step I made use of the `corrmatrix` function. Then Codacy complained about it being a method unnecessarily since there is no use of `self`. That's why I refactored it to being a function outside of the class. I think, that I already did refactor the whole of the class in the `prepare_major_release_2.0.0` . Here I just wanted to satisfy Codacy with minimal effort. The renaming still seemed reasonable to me.

Reply:
Resolved and approved via telephone just now.